### ISSUE TITLE
Can't receive push notification even on device being shown subscribed on onesignal dashboard.

### ISSUE THREAD

**Issue by KANE-99** [OneSignal/OneSignal-Android-SDK#1272]
**Description:**

Install two apps with same OneSignal app id. When the latter installed app is uninstalled, OneSignal shows device as subscribed in dashboard but we try to send push notification....it fails and device becomes unsubscribed on OneSignal dashboard. But when we open the remaining app on device, it becomes subscribed again on OneSignal dashboard but the push notifications keep failing.

**Environment**
OneSignal SDK 4.0.0

**Steps to Reproduce Issue:**

1. Add the OneSignal SDK to two dummy projects.
1. Give same App ID to both projects.
1. Install both apps.
1. Uninstall latter installed app.
1. Attempt to receive push notification

**Anything else:**

On opening the former app after latter app is uninstalled, below is the generated stack trace, as we can see that new device token is generated, but push notification fails. 

`
Device registered, push token = fzaRUnMzSyGUsNrd8IduL5:APA91bEzTiKuSz5DTGKBB8dBnGhZNUTFhmrPdlDXyKhPcW6L5ulcBwSiI8CDjkndkSm5oBw90qVUQ0-eENAmSZIb6beUkj9miPE-frcsM_p9datRikcQWeBLKe_tzLsf5DobXOfsobTQ
2021-02-01 15:54:19.644 6270-6325/com.example.nextapp D/OneSignal: registerForPushToken completed with id: fzaRUnMzSyGUsNrd8IduL5:APA91bEzTiKuSz5DTGKBB8dBnGhZNUTFhmrPdlDXyKhPcW6L5ulcBwSiI8CDjkndkSm5oBw90qVUQ0-eENAmSZIb6beUkj9miPE-frcsM_p9datRikcQWeBLKe_tzLsf5DobXOfsobTQ status: 1
2021-02-01 15:54:19.647 6270-6325/com.example.nextapp D/OneSignal: registerUser:registerForPushFired:true, locationFired: true, remoteParams: com.onesignal.OneSignalRemoteParams$2@c8aeaf, appId: cbc14ac8-63c5-4003-9fef-dde7ebada477
2021-02-01 15:54:19.825 6270-6338/com.example.nextapp D/OneSignal: registerUserTask calling readyToUpdate
2021-02-01 15:54:24.834 6270-6324/com.example.nextapp D/OneSignal: UserStateSynchronizer internalSyncUserState from session call: true jsonBody: {"app_id":"cbc14ac8-63c5-4003-9fef-dde7ebada477"}
2021-02-01 15:54:24.836 6270-6354/com.example.nextapp D/OneSignal: OneSignalRestClient: Making request to: https://api.onesignal.com/players/8bd8b895-4010-4f64-acf0-f1aab38730e2/on_session
2021-02-01 15:54:24.837 6270-6354/com.example.nextapp D/OneSignal: OneSignalRestClient: POST SEND JSON: {"app_id":"cbc14ac8-63c5-4003-9fef-dde7ebada477"}
2021-02-01 15:54:25.049 6270-6354/com.example.nextapp V/OneSignal: OneSignalRestClient: After con.getResponseCode to: https://api.onesignal.com/players/8bd8b895-4010-4f64-acf0-f1aab38730e2/on_session
2021-02-01 15:54:25.050 6270-6354/com.example.nextapp D/OneSignal: OneSignalRestClient: Successfully finished request to: https://api.onesignal.com/players/8bd8b895-4010-4f64-acf0-f1aab38730e2/on_session
2021-02-01 15:54:25.054 6270-6354/com.example.nextapp D/OneSignal: OneSignalRestClient: POST RECEIVED JSON: {"success":true,"id":"8bd8b895-4010-4f64-acf0-f1aab38730e2"}
2021-02-01 15:54:25.057 6270-6355/com.example.nextapp D/OneSignal: doCreateOrNewSession:response: {"success":true,"id":"8bd8b895-4010-4f64-acf0-f1aab38730e2"}
2021-02-01 15:54:25.061 6270-6355/com.example.nextapp I/OneSignal: Device registered, UserId = 8bd8b895-4010-4f64-acf0-f1aab38730e2
`

**Comment by bobbyirawan09** [comment]
I'm also facing the same issue, is there's some kind of bug that comes with the v4.0.0? I tried on version 3.x.x and it seems fine, hope it will only affect device that installs more than one app with the same OneSignal app id

**Comment by KANE-99** [comment]
Yes, It only affects devices with same OneSignal app id. Hope this issue gets fixed as currently we have to handle this by manually editing device using OneSignal api.

**Comment by jkasten2** [comment]
@KANE-99 @bobbyirawan09 
> Install two apps with same OneSignal app id. 

OneSignal is designed with a 1-to-1 app (Android package name) and OneSignal app in mind. So for each different app you have on the Google Play store you should have a different OneSignal app_id.

> When the latter installed app is uninstalled, OneSignal shows device as subscribed in dashboard but we try to send push notification....it fails and device becomes unsubscribed on OneSignal dashboard. But when we open the remaining app on device, it becomes subscribed again on OneSignal dashboard but the push notifications keep failing.

Providing some more detail to this of how OneSignal is working under the hood. What is most likely happening here is that both app on the device are sharing the same OneSignal player (AKA OneSignal subscription) but each app will have it's own push token (FCM token). This is due to the OneSignal SDK detecting the device is the same based of the Google Ad Id (this however was removed from this [SDK 4.3.4 on 2021/05/06](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/4.3.4))

In summary even though the specific issue may no longer be an issue as of version 4.3.4 we still recommend you make a 2nd OneSignal app for your other app to avoid any other possible issues.
